### DISCUSSION TITLE
Why isn't winget the default installation method?

### DISCUSSION THREAD

**Discussion by lambdakilo** [coreybutler/nvm-windows#1021] [General] [answered]
The README suggests downloading and installing through releases. What's the reason why the README doesn't just say
```powershell
winget install nvm-windows
```
?

**Comment by coreybutler** [comment, marked as answer]
First, winget doesn't ship with Windows. We don't need to arbitrarily add a dependency. In many enterprises, winget isn't allowed at all, which would create a blocking barrier for nearly half of the NVM4W community. Bottom line: winget is a convenience, not a default. We don't support it, but someone is maintaining a winget release.

With [Runtime](https://github.com/coreybutler/nvm-windows/wiki/Runtime), we will release on several distribution channels. Winget is one of the channels we'd like to release on.

**Reply by lambdakilo** [reply]
I'll mark this as an answer as this might be the closest thing I'll ever get to an answer.

With a quick search winget does ship with Windows. I don't think winget is a dependency but a distribution platform. Those enterprises that don't allow the use of winget should maybe think twice if they allow downloading arbitrary binaries from a browser rather than from winget. Bottom line: winget should be a default. We should encourage enterprises and other entities to download software the modern and safer way.

**Reply by coreybutler** [reply]
Winget doesn't ship with Windows by default. It's available as an add-on in the Windows Store, as described in the [Microsoft Documentation](https://learn.microsoft.com/en-us/windows/package-manager/winget/). If NVM4W required winget to be installed, then NVM4W would be dependent on the existence of winget, thus making it a dependency to acquire NVM4W.

Most organizations do not allow arbitrary downloads either. Randomly downloading an executable will not work in typical enterprises. NVM4W is the only Node version manager that code-signs its installers and executables with an extended validation certificate (i.e. we've gone through the trust-verification process for Microsoft). In other words, it's not just a random download, it is a vetted and trusted application. This is why it is allow-listed by thousands of large enterprises.

Winget is no safer than downloading arbitrary code. The Winget registry is an open one, meaning anyone, including bad actors, may contribute malicious applications. The modern safe way for enterprises to install software is through Active Directory. Organizations test/vet software and release it through group policies defined specifically for their organizations. This is why NVM4W ships a "noinstall" version, which has been verified with all of the modern antivirus organizations. Most enterprises use this version to create their own Active Directory deployments. This allows them to install/update applications on employee computers automatically... meaning users don't perform any installations on their own (the software just shows up on their computer).

I'll reiterate that in my opinion, winget is a convenience for developers who are not on Active Directory and seek an experience similar to Homebrew. That's why we're planning to support it in Runtime, but we will still recommend Active Directory deployments as the preferred mechanism for enterprise installations. In that vein, we are planning to switch from an exe-based installer to an MSI-based installer to make this process easier for enterprises. We will continue to code-sign and verify Runtime with antivirus vendors as we have with NVM4W.

**Reply by lambdakilo** [reply]
Thanks for the reply.

**Comment by CherryDT** [comment]
> winget doesn't ship with Windows

That's no longer true, by the way. It is now installed by default.

**Reply by Remzi1993** [reply]
Only on Windows 11 and since almost half of the users still use Windows 10 (as of this moment - 22-04-2025) therefore it's not a good idea. Maybe when Windows 10 reached end of support and 90%+ of users use Windows 11 or higher then it would make sense to use the default package manager of an operating system.

**Reply by CherryDT** [reply]
It is also by default on Windows 10:

> Winget is available on Windows 10 version 1809 (October 2018 Update) or later. **It comes standard with Windows 10 May 2020 Update (version 2004) and later versions.**

**Reply by Remzi1993** [reply]
> It is also by default on Windows 10:
> 
> > Winget is available on Windows 10 version 1809 (October 2018 Update) or later. **It comes standard with Windows 10 May 2020 Update (version 2004) and later versions.**

Maybe after upgrades and updates, but Winget is not a default in older versions/builds of Windows 10. Not everyone is on the latest stable build, especially enterprise.
